solr/core/src/java/org/apache/solr/search/facet/FacetModule.java

```diff
@@ -319,7 +323,10 @@ public void handleResponses(ResponseBuilder rb, ShardRequest sreq) {
         // System.err.println("REFINE FACET RESULT FROM SHARD = " + facet);
         // call merge again with a diff flag set on the context???
         facetState.mcontext.root = facet;
-        facetState.mcontext.setShard(shardRsp.getShard()); // TODO: roll newShard into setShard?
+        if (!facetState.mcontext.setShard(shardRsp.getShard())) { // TODO: roll newShard into setShard?
```

Comment:
I could also be convinced to do what this TODO says and actually just fold in `newShard` as needed when `setShard` is called.